lib/plugins/aws/deploy/compile/events/stream/index.js

```diff
@@ -78,8 +88,15 @@ class AwsCompileStreamEvents {
                 .Error(errorMessage);
             }
 
-            const streamType = EventSourceArn.split(':')[2];
-            const streamName = EventSourceArn.split('/')[1];
+            const streamType = event.stream.type || EventSourceArn.split(':')[2];
+            const streamName = (function () {
+              if (EventSourceArn['Fn::GetAtt']) {
+                return EventSourceArn['Fn::GetAtt'][0];
+              } else if (EventSourceArn['Fn::ImportValue']) {
+                return EventSourceArn['Fn::ImportValue'];
```

Comment:
Me either.  I started to think about what would be a solution before my attention was seized elsewhere.  Truncation seems necessary.  Then the worry becomes that previously unique values come into conflict.  To avoid that, you could take the windows ~N suffix approach but then the framework would have to keep track of the mappings in order to maintain consistency across changes to the template that reordered conflicting mappings.  As such, I'd say truncate and leave a little bit of a documentation trail in the streams docs.  If conflicts occur then we can advise users on what the issue is and if this corner case actually occurs and becomes a blocker for users, then we can tackle (or guide interested parties in tackling) it then.